editor/plugins/skeleton_editor_plugin.cpp

```diff
@@ -30,13 +30,326 @@
+	}
+
+	for (int i = 0; i < 12; ++i) {
+		transform[i] = memnew(EditorSpinSlider());
+		transform[i]->set_label(desc[i]);
+		setup_spinner(transform[i], true);
+		transform_grid->add_child(transform[i]);
+	}
+}
+
+void BoneTransformEditor::setup_spinner(EditorSpinSlider *spinner, const bool is_transform_spinner) {
+	spinner->set_flat(true);
+	spinner->set_min(-10000);
+	spinner->set_max(10000);
+	spinner->set_step(0.001);
+	spinner->set_hide_slider(true);
+	spinner->set_allow_greater(true);
+	spinner->set_allow_lesser(true);
+	spinner->set_h_size_flags(SIZE_EXPAND_FILL);
+
+	spinner->connect_compat("value_changed", this, "_value_changed", varray(is_transform_spinner));
+}
+
+void BoneTransformEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			create_editors();
+			key_button->connect_compat("pressed", this, "_key_button_pressed");
+			enabled_checkbox->connect_compat("toggled", this, "_checkbox_toggled");
+
+			FALLTHROUGH;
+		}
+		case NOTIFICATION_THEME_CHANGED: {
+			const Color base = get_color("accent_color", "Editor");
+			const Color bg_color = get_color("property_color", "Editor");
+			const Color bg_lbl_color(bg_color.r, bg_color.g, bg_color.b, 0.5);
+
+			for (int i = 0; i < 12; i++) {
+				Color c = base;
+				c.set_hsv(float(i % 3) / 3.0 + 0.05, c.get_s() * 0.75, c.get_v());
+
+				if (i < 3) {
+					if (translation[i] != nullptr)
+						translation[i]->set_custom_label_color(true, c);
+					if (rotation[i] != nullptr)
+						rotation[i]->set_custom_label_color(true, c);
+					if (scale[i] != nullptr)
+						scale[i]->set_custom_label_color(true, c);
+				}
+
+				if (transform[i] != nullptr) {
+					transform[i]->set_custom_label_color(true, c);
+				}
+			}
+			break;
+		}
+		case NOTIFICATION_SORT_CHILDREN: {
+			const Ref<Font> font = get_font("font", "Tree");
+
+			Point2 buffer;
+			buffer.x += get_constant("inspector_margin", "Editor");
+			buffer.y += font->get_height();
+			buffer.y += get_constant("vseparation", "Tree");
+
+			const float vector_height = translation_grid->get_size().y;
+			const float transform_height = transform_grid->get_size().y;
+			const float button_height = key_button->get_size().y;
+
+			const float width = get_size().x - get_constant("inspector_margin", "Editor");
+
+			if (keyable && section->get_vbox()->is_visible()) {
+				background_rects[0] = Rect2(key_button->get_position() + buffer, Size2(width, button_height));
+			} else {
+				background_rects[0] = Rect2(0, 0, 0, 0);
+			}
+
+			if (section->get_vbox()->is_visible()) {
+				background_rects[1] = Rect2(translation_grid->get_position() + buffer, Size2(width, vector_height));
+				background_rects[2] = Rect2(rotation_grid->get_position() + buffer, Size2(width, vector_height));
+				background_rects[3] = Rect2(scale_grid->get_position() + buffer, Size2(width, vector_height));
+				background_rects[4] = Rect2(transform_grid->get_position() + buffer, Size2(width, transform_height));
```

Comment:
A for loop would be better here

Reply:
You cannot iterate here, because it does different things for the 5 controls.

Reply:
I guess I can make it use a vector of vec2.